tests/units/online/test_advanced_goto_functions.py

```diff
@@ -890,3 +890,26 @@ def test_head_rotation(reachy_sdk_zeroed: ReachySDK) -> None:
 
     with pytest.raises(ValueError):
         reachy_sdk_zeroed.head.rotate_by(roll=30, interpolation_mode="coucou")
+
+
+@pytest.mark.online
+def test_waiting_goto(reachy_sdk_zeroed: ReachySDK) -> None:
+    reachy_sdk_zeroed.goto_posture("elbow_90", duration=1, wait=True)
+    for arm in [reachy_sdk_zeroed.l_arm, reachy_sdk_zeroed.r_arm]:
+        x, y, z = arm.forward_kinematics()[:3, 3]
+        reachy_sdk_zeroed.head.look_at(x=x, y=y, z=z, duration=1, wait=True)
+        arm.translate_by(x=0.2, y=0.0, z=0.1, wait=False)
+        arm.translate_by(x=-0.2, y=0.0, z=-0.1, wait=False)
+        arm.translate_by(x=0.2, y=0.0, z=0.1, wait=False)
+        last_gotoid = arm.translate_by(x=-0.2, y=0.0, z=-0.1, wait=False)
+        while not reachy_sdk_zeroed.is_goto_finished(last_gotoid):
+            x, y, z = arm.forward_kinematics()[:3, 3]
+            reachy_sdk_zeroed.head.look_at(x=x, y=y, z=z, duration=0.005, wait=True)
+
+    reachy_sdk_zeroed.goto_posture("elbow_90", duration=1, wait=True)
```

Comment:
This one actually hangs without the timeout. Maybe can keep only this part?

Usually you want to have an assert in your test something.
I would move just one arm a bit and test if the target position is reached. Like the smallest for loop needed to create the bug without the timeout, and with a long enough movement to check the new position.